cmd/osm-bootstrap/osm-bootstrap_test.go

```diff
@@ -218,57 +283,182 @@ func TestCreateDefaultMeshConfig(t *testing.T) {
-				meshConfigClient: tc.meshConfigClient,
-				namespace:        tc.namespace,
+				kubeClient:   tc.kubeClient,
+				configClient: tc.configClient,
+				namespace:    tc.namespace,
 			}
 
 			err := b.ensureMeshConfig()
 			if tc.expectErr {
 				assert.NotNil(err)
 			} else {
 				assert.Nil(err)
-				config, err := b.meshConfigClient.ConfigV1alpha2().MeshConfigs(b.namespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
+				config, err := b.configClient.ConfigV1alpha2().MeshConfigs(b.namespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
+				assert.Nil(err)
+				assert.Contains(config.Annotations, "kubectl.kubernetes.io/last-applied-configuration")
+			}
+		})
+	}
+}
+
+func TestCreateMeshRootCertificateConfig(t *testing.T) {
+	tests := []struct {
+		name                             string
+		namespace                        string
+		kubeClient                       kubernetes.Interface
+		configClient                     configClientset.Interface
+		expectDefaultMeshRootCertificate bool
+		expectErr                        bool
+	}{
+		{
+			name:                             "successfully create default MeshRootCertificate from preset configmap",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset([]runtime.Object{testPresetMeshRootCertificate}...),
+			configClient:                     fakeConfig.NewSimpleClientset(),
+			expectDefaultMeshRootCertificate: true,
+			expectErr:                        false,
+		},
+		{
+			name:                             "preset configmap does not exist",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset(),
+			configClient:                     fakeConfig.NewSimpleClientset(),
+			expectDefaultMeshRootCertificate: false,
+			expectErr:                        true,
+		},
+		{
+			name:                             "default MeshRootCertificate already exists",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset([]runtime.Object{testPresetMeshRootCertificate}...),
+			configClient:                     fakeConfig.NewSimpleClientset([]runtime.Object{testMeshRootCertificate}...),
+			expectDefaultMeshRootCertificate: true,
+			expectErr:                        false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert := tassert.New(t)
+			b := bootstrap{
+				kubeClient:   tc.kubeClient,
+				configClient: tc.configClient,
+				namespace:    tc.namespace,
+			}
+
+			err := b.createMeshRootCertificate()
+			if tc.expectErr {
+				assert.NotNil(err)
+			} else {
+				assert.Nil(err)
+			}
+
+			_, err = b.configClient.ConfigV1alpha2().MeshRootCertificates(b.namespace).Get(context.TODO(), meshRootCertificateName, metav1.GetOptions{})
+			if tc.expectDefaultMeshRootCertificate {
+				if err == nil {
+					assert.Nil(err)
+				}
```

Comment:
Not sure I follow exactly what the intent is here, but this assert can never fail.

Reply:
Good catch, thank you! This was copied from the similar Mesh Config test. I've gone through the other tests and cleaned up some of the checks.